_posts/2020-03-24-dask-developer-workshop.markdown

```diff
@@ -0,0 +1,90 @@
+The Data Engineering team leverages the distributed scheduler for parallel task execution during the data transformation steps between database and Parquet datasets. Most parts of these pipelines use Dask bags or delayed objects for map/reduce operations. Dask dataframes are especially useful when we reshuffle datasets with an existing partitioning to a differently partitioned dataset. We use [Kartothek](https://github.com/JDASoftwareGroup/kartothek) for this purpose.
+
+A lot of Dask users we met at the workshop also utilized Dask clusters for data-heavy calculations. However, the Blue Yonder use case remained special: most other users were working in an R&D-like environment. Their computations are not running in comparable production environments. Most importantly, these users had no service level agreements with customers about, for example, delivery times for predictions. It seems that Blue Yonder puts higher requirements on the stability and robustness of the distributed scheduler than any other of the users present at the conference.
+
+## Issues we have encountered with Dask and potential improvements
+
+In these discussions we talked about a lot of topics, some of which have been or still are big problems for us. We encountered community members who had similar experiences. Below are some of the "rants" we want to share.
+
+
+### Distributed stability
+We migrated some of our largest data pipelines to Dask in the last months of 2019. For these pipelines, we started to observe significant instability during the computation of the Dask graph. We hadn't seen this issue in any of our previous pipelines running on distributed. Our team contributed several patches upstream in order to resolve these issues on our side. 
+ 
+During the working sessions, a number of conversations took place related to the stability of the distributed scheduler, involving Dask developers and users.
+One idea which came up to increase the overall robustness of distributed was _replication of task results_. The idea is the following: if a worker which is holding a task result is shut down, but the task result is replicated, then we do not need to re-compute parts of the graph again.
+As of the time of this writing, we have not had a chance to actively work on something like this yet, but it is something that we keep in mind.
+
+### Performance and graph optimization
+While monitoring the execution of one of our Dask production pipelines, we observed that the amount of memory consumed by the workers was significantly higher than one would expect in an ideal scenario where tasks are executed in a way which minimizes memory usage of the workers.
+When investigating the Dask.optimization module, we saw that code to optimize memory usage was already there. In practice, however, the graphs are not executed in such an order because of other constraints during execution.
```

Comment:
"In practice, however, the graphs are not executed in such an order because of other constraints during execution" - what does such an order here refer to?

Reply:
Optimal order in terms of memory usage

Reply:
```suggestion
When investigating the Dask.optimization module, we saw that code to optimize memory usage was already there. In practice, however, the graphs are not executed in the optimal order with regards to memory usage because of other constraints during execution.
```